SimPEG/utils/io_utils/io_utils_electromagnetics.py

```diff
@@ -1,35 +1,208 @@
 from __future__ import print_function
 import numpy as np
 from discretize.utils import mkvc
+import warnings
 
 
 ########################################################################################
 #                  DIRECT CURRENT RESISTIVITY AND INDUCED POLARIZATION
 ########################################################################################
 
-
-def read_dcip3d_ubc(file_name, data_type):
+def read_dcip_xyz(
+    file_name,
+    data_type,
+    a_headers=['XA', 'YA', 'ZA'],
+    b_headers=['XB', 'YB', 'ZB'],
+    m_headers=['XM', 'YM', 'ZM'],
+    n_headers=['XN', 'YN', 'ZN'],
+    data_header=None,
+    uncertainties_header=None,
+    dict_headers=None,
+    is_surface_data = False
+):
     """
-    Read UBC DCIP3D formatted survey, predicted or observed data files.
+    Read XYZ formatted file for 2D or 3D formatted DC/IP data.
+    
+    Parameters
+    ----------
+    
+    file_name: Path to xyz file
+    data_type: ["volt", "apparent_resistivity", "apparent_chargeability"]
+    a_headers: A list or tuple of strings representing the headers of the A-electrode location
+    b_headers: A list or tuple of strings representing the headers of the B-electrode location
+    m_headers: A list or tuple of strings representing the headers of the M-electrode location
+    n_headers: A list or tuple of strings representing the headers of the N-electrode location
+    data_header: String representing the header for the data column
+    uncertainties_header: String representing the header for the data column
+    dict_headers: list or tuple of strings for additional data columns being loaded into a dictionary
+    is_surface_data: True if surface formatted and electrode elevations are not supplied
+    """
+    
+    if data_type not in ["volt", "apparent_resistivity", "apparent_chargeability"]:
+        raise Exception(
+            "data_type must be one of 'volt', 'apparent_resistivity', 'apparent_chargeability'"
+        )
+    
+    # Prevent circular import
+    from ...electromagnetics.static import resistivity as dc
+    from ...data import Data
+    
+    # Load file headers
```

Comment:
could you use panda's to load all of these in simpler?

Reply:
I don't have much familiarity with pandas. Would it that much of an improvement?